content/post/000-ipfs-in-web-browsers.md

```diff
@@ -0,0 +1,74 @@
+* UDP sockets enable DNS-based service discovery of go-ipfs in LAN, and we are working on browser to browser discovery methods that work in offline environments
+
+
+By default, our browser extension still expects [IPFS Desktop](https://github.com/ipfs-shipyard/ipfs-desktop#ipfs-desktop) to be installed, however you can select the experimental option shown below.
+
+![IPFS configuration option for embedded node using Chrome sockets API in Brave](https://file.globalupload.io/g3217mupyf.png)
+
+## Opera
+
+Opera is another [early leader](https://blogs.opera.com/mobile/2018/12/ofa-49-with-crypto/) in the space. They've developed *native* crypto wallets in their browsers, which began shipping in 2018 in Android and now released in their iOS and desktop browsers as well.
+
+Earlier this year [Opera announced they were working on support for IPFS](https://twitter.com/ensdomains/status/1102884419017297921)! We've been talking ever since, and are looking forward to IPFS support in their Android browser by the end of this year.
+
+![Opera announcing ENS and IPFS at EthCC March 2019](https://file.globalupload.io/fE5aG9KKAr.png)
+
+## The Work is Just Beginning
+
+Integrated IPFS gateway support is one big step forward, but as the name implies, it isn't truly decentralized. Only when a browser can be a full participant in the network - both a client *and* a server - will we have a truly distributed web. 
+
+These early-adopters light the way, proving out what works and doesn't and easing development for dapp developers in the near term.
+
+But there's a lot of work to be done. IPFS  itself is still in flux, with major work ongoing in performance and robustness, and feature work around privacy and content filtering.
+
+We're working on detailed use-cases to share with standards bodies, and a threat model that browser vendors can build and design against. Both are prerequisites for a future where all major browsers have interoperable implementations, which is what would make IPFS truly be part of the web.
+
+Web browsers have been solely a client for over 30 years - putting all power in the hands of the server. This year marks a paradigmatic shift in the belief of what a browser can be. Change is now happening at the level from which that power is derived: the network.
+
+## This is a Preview!
+
+Consider this a sneak peek. There's more on the way:
+
+* In Brave, the gateway is, well... a gateway to even more: We're working on local discovery (mDNS) of IPFS nodes, and also Brave-to-Brave connectivity for embedded nodes.
+* We're continuing our experiments with [libdweb](https://github.com/mozilla/libdweb) in Firefox, both on desktop and Android. 
+* Puma browser, the first mobile browser with support for [Coil's micropayments for publishers](https://www.grantfortheweb.org/), is interested in IPFS support. If you're a mobile developer interested in this work, [contact me](mailto:dietrich@protocol.ai)!
```

Comment:
Nice! Maybe put out a list though? Otherwise you might start getting a lot of spam emails...

Reply:
fixed